src/support/intervals.cpp

```diff
@@ -36,26 +37,24 @@ IntervalProcessor::filterOverlaps(std::vector<Interval>& intervals) {
 
   std::sort(
     intIntervals.begin(), intIntervals.end(), [](const auto& a, const auto& b) {
-      return a.first.start < b.first.end;
+      return a.first.start < b.first.start;
```

Comment:
No need for the lambda here if you fix `operator<` to be a total order (meaning that for any pair of intervals `a` and `b`, exactly one of `a < b`,`b < a`, or `a == b` is true)

Reply:
done